src/libraries/System.Runtime/tests/System.Runtime.Tests/System/ActivatorTests.cs

```diff
@@ -107,6 +107,65 @@ public void CreateInstance_PublicOnlyTypeWithPrivateDefaultConstructor_ThrowsMis
             Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(TypeWithPrivateDefaultConstructor), nonPublic: false));
         }
 
+        [Fact]
+        public void CreateInstance_WithCustomBinder_ThrowsMissingMethodException()
+        {
+            // MissingMethodException not caused by a binder must not contain an inner exception.
+            var mme = Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(TypeWithPrivateDefaultConstructor), nonPublic: false));
+            Assert.Contains("System.Tests.ActivatorTests.TypeWithPrivateDefaultConstructor", mme.Message);
+            Assert.Null(mme.InnerException);
+
+            // MissingMethodException caused by a binder must be wrapped.
+            mme = Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(
+                typeof(object), BindingFlags.CreateInstance,
+                new CustomBinder() { BindToMethodAction = () => throw new MissingMethodException("Hello, World!!") },
+                null, null, null
+            ));
+            Assert.Contains("System.Object", mme.Message);
+            Assert.NotNull(mme.InnerException);
+            Assert.IsType<MissingMethodException>(mme.InnerException);
+            Assert.StrictEqual("Hello, World!!", mme.InnerException.Message);
```

Comment:
To avoid build errors:

```suggestion
            Assert.Equal("Hello, World!!", mme.InnerException.Message);
```